doc/connectivity/bluetooth/api/l2cap.rst

```diff
@@ -3,11 +3,42 @@
 Logical Link Control and Adaptation Protocol (L2CAP)
 ####################################################
 
-L2CAP layer enables connection-oriented channels which can be enable with the
-configuration option: :kconfig:option:`CONFIG_BT_L2CAP_DYNAMIC_CHANNEL`. This channels
+L2CAP layer enables connection-oriented channels which can be enabled with the
+configuration option: :kconfig:option:`CONFIG_BT_L2CAP_DYNAMIC_CHANNEL`. These channels
 support segmentation and reassembly transparently, they also support credit
 based flow control making it suitable for data streams.
 
+The user can also define fixed channels using the :c:macro:`BT_L2CAP_FIXED_CHANNEL_DEFINE`
+macro. Fixed channels are initialized upon connection, and do not support segmentation. An example
+of how to define a fixed channel is shown below.
+
+.. code-block:: c
+
+   static struct bt_l2cap_chan fixed_chan[CONFIG_BT_MAX_CONN] = {0};
```

Comment:
```suggestion
   static struct bt_l2cap_chan fixed_chan[CONFIG_BT_MAX_CONN];
```

This happens automatically. The Zephyr linter forbids a zero initializer on static actually.

[I meant](https://github.com/zephyrproject-rtos/zephyr/pull/95714#discussion_r2348361022) initialization must be done in the accept callback. This is so it's initialized for every new connection.

I see it like this: Logically, `fixed_chan` is a pool of memory for potential `struct bt_l2cap_chan`. The `bt_l2cap_chan` object is *instantiated* in accept(), so it need to be initialized there.

The reason we access the pool by `bt_conn_index` is that it's a allocation scheme that is very simple and perfect based on the fact that there can only be one fixed channel with a particular CID per connection.

Reply:
Initialization can be done by compound literals or memset.

I prefer compound literals because it's type checked, preventing mistakes in sizeof expressions and pointer pointers.

Reply:
If it is already zero-initialized, I don't see the need to clutter the example with additional initialization? If anything, we could add a comment about this to the accept callback, but then I'd prefer if you add a suggestion about this there.

Reply:
OK. I'll move the discussion there. Here, just remove the ` = {0}`.